package/batocera/core/batocera-splash/S29abortsplash

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+# Stop Splash Screen
+
+[[ $1 == start ]] || exit 0
+
+# We can set here an additional timer to extend videoplay
+# This value can be setted to batocera.conf
+splash_autoplay ()
+{
+    count=0
+    while [[ $count -lt 90 ]]
+    do
+        sleep 1
+        ((count++))
+        kill -0 $PID &>/dev/null  || { echo "Video finished"; break; }
+    done
+}
+
+splash_default ()
+{
+    # Stop the video when ready
+    # Wait for emulationstation or Kodi, but not more than 20 seconds
+    count=0
+    while [[ ! -f "/tmp/emulationstation.ready" && ! -e "/var/run/kodi.msg" && $count -lt 20 ]]; do
+        sleep 1
+        ((count++))
+    done
+}
+
+#### MAIN
+video=0; picture=0
+if [[ -f /tmp/videoplayer.pid ]]; then
+    video=1
+    PID=$(cat /tmp/videoplayer.pid)
+else
+    picture=1
+fi
+
+splash_setting=$(batocera-settings -command load -key splash.screen.length)
+case ${splash_setting,,} in
+
+    def*)
+        # Standard BATOCERA setting, fastest boot possible
+        # A video splash will be displayed ~16s on Pi3 platform
+        #
+        splash_default &
+        [[ $video -eq 1 ]] && kill -9 $PID
+    ;;
+
+    auto)
+        # Play video as long as possible, if you are using png -> default
+        #
+        [[ $picture -eq 1 ]] && splash_default &
+        [[ $video -eq 1 ]] && splash_autoplay && kill -9 $PID
```

Comment:
why a second kill?

Reply:
There is no seconds kill. `kill -0` just checks if the process is still alive

Reply:
ok